lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -128,6 +131,75 @@ class ReportUIFeatures {
     dropdown.addEventListener('click', this.onExport);
   }
 
+  _setupThirdPartyFilter() {
+    const pageTLDPlusOne = new URL(this.json.finalUrl).origin.split('.').slice(-2).join('.');
+
+    // get all tables with a text url
+    /** @type {Array<HTMLTableElement>} */
+    const tables = Array.from(this._document.querySelectorAll('.lh-table'));
+    tables.filter(el => el.querySelector('td.lh-table-column--url'))
+      .filter(el => /** @type {Element} */ (el.closest('.lh-audit')).id !== 'uses-rel-preconnect')
+      .forEach((el, index) => {
+        /** @type {NodeListOf<HTMLElement>} */
+        const urlItems = el.querySelectorAll('.lh-text__url');
+        /** @type {Record<string, HTMLTableRowElement>} */
+        const thirdPartyRows = {};
```

Comment:
we can pull the first 10ish lines of this out into a `const thirdPRowElems = this._getThirdPartyRows(el, finalUrl);`